client/src/api.js

```diff
@@ -0,0 +1,547 @@
+
+    return ({
+      title: userName.name,
+      start: newDate,
+      backgroundColor: day.status === 'REJECTED' ? 'red' : 'green'
+    })
+  })
+  } else {
+      switch (response.status) {
+        case 400:
+          throw new Error('Bad request. Check query parameters.')
+        case 401:
+          window.location.replace("http://localhost:3000/login")
+        case 403:
+          window.location.replace("http://localhost:3000/login")
+        case 404:
+          throw new Error('User with given ID does not exist.')
+        case 500:
+          throw new Error('Internal server error.')
+        default:
+          throw new Error(response.statusText)
+      }
+    }
+}
+
+// ******************** LOAD DATA to CALENDAR - EMPLOYER ********************
+
+export const getAdminCalendar = async () => {
+
+  let response;
+
+  try {
+    response = await fetch(
+      `${window.config.baseUrl}/users/requests/vacation?status=ACCEPTED`, {
+        headers: {
+          'Accept': 'application/json',
+          Authorization: 6
+        },
+        credentials: 'include',
+        method: 'GET',
+      }
+    );
+  } catch (e) {
+    throw 'Server is not available'
+  }
+
+  if (response.ok) {
+    const data = await response.json();
+    
+    return data.map(day => {
+
+      const newDate = day.date.split("/").join("-");
+
+      return ( {
+      title: day.firstName + ' ' + day.lastName,
+      start: newDate
+      })
+    })
+  } else {
+      switch (response.status) {
+        case 400:
+          throw new Error('Bad request. Check query parameters.')
+        case 401:
+          window.location.replace("http://localhost:3000/login")
+        case 403:
+          window.location.replace("http://localhost:3000/login")
+        case 500:
+          throw new Error('Internal server error.')
+        default:
+          throw new Error(response.statusText)
+      }
+    }
+}
+
+// ******************** ADD EVENT to CALENDAR - EMPLOYEE ********************
+
+export async function addEventApi(userName, dataAddEventEmployee) {
+  let response;
+
+  try {
+  // send accepted request to server
+    response = await fetch(`${window.config.baseUrl}/user/calendar/create`, {
+      headers: {
+        Authorization: 6,
+        'Content-Type': 'application/json',
+      },
+      credentials: 'include',
+      method: 'POST',
+  // object which is sent to server
+  body: JSON.stringify(dataAddEventEmployee),
+    });
+  } catch (e) {
+    throw 'Server is not available'
+  }
+
+  if (response.ok) {
+      
+    response = await fetch(
+    `${window.config.baseUrl}/users/requests/vacation?status=PENDING`, {
+      headers: {
+        Authorization: 1
+      },
+      credentials: 'include',
+    });
+    const data = await response.json();
+      
+    return data.map(request => {
+      const a = request.date;
+      const b = [a.slice(0, 4), "-", a.slice(5, 7), "-", a.slice(8, 10)].join('');
```

Comment:
We found a smarter way to do this conversion: `const b = request.date.split("/").join("-")`.

Also `a` and `b` are horrible variable names.